app/src/ui/lib/list/list.tsx

```diff
@@ -1067,10 +1230,91 @@ export class List extends React.Component<IListProps, IListState> {
         aria-label={this.props.ariaLabel}
       >
         {content}
+        {this.renderKeyboardInsertionElement()}
       </div>
     )
   }
 
+  private renderKeyboardInsertionElement = () => {
+    const { keyboardInsertionData, keyboardInsertionElementRenderer } =
+      this.props
+    const { keyboardInsertionIndexPath } = this.state
+
+    if (!keyboardInsertionData || !keyboardInsertionIndexPath) {
+      return null
+    }
+
+    if (!this.list) {
+      return
+    }
+
+    const listRect = this.list.getBoundingClientRect()
+
+    const renderOverlay = (
+      top: number | string,
+      left: number | string,
+      width: number | string,
+      height: number | string
+    ) => {
+      return (
+        <div
+          style={{
+            top,
+            left,
+            width,
+            height,
+          }}
+          className="overlay"
+        />
+      )
+    }
```

Comment:
You could make this quite a bit more compact using a temporary type if you wanted to

```suggestion
    type U = number | string
    const renderOverlay = (top: U, left: U, width: U, height: U) => (
      <div style={{ top, left, width, height }} className="overlay" />
    )
```